packages/ui/src/components/editor/properties/mesh/index.tsx

```diff
@@ -37,10 +38,37 @@ import { Material } from 'three'
 import Accordion from '../../../../primitives/tailwind/Accordion'
 import GeometryEditor from './geometryEditor'
 
+const MaterialIsInAuthoringLayer = (material: Material): boolean => {
+  return !!UUIDComponent.getEntityByUUID(material.uuid as EntityUUID, Layers.Authoring)
+}
+
 const MeshNodeEditor: EditorComponentType = (props: { entity: Entity }) => {
   const entity = props.entity
   const { t } = useTranslation()
   const meshComponent = getComponent(entity, MeshComponent)
+
+  const materialEditor = () => {
+    if (Array.isArray(meshComponent?.material) && meshComponent.material.length > 0) {
+      return meshComponent.material.map((material, index) => {
+        if (!MaterialIsInAuthoringLayer(material)) return null
+        return (
+          <Accordion title={t('editor:properties.mesh.materialEditor')} key={index}>
```

Comment:
we should key by data instead of list index, such that if something is unshifted, the key still points to the correct child - should help with implicit memoization